core/src/main/java/it/pagopa/selfcare/external_api/core/OnboardingServiceImpl.java

```diff
@@ -242,7 +200,9 @@ public void autoApprovalOnboarding(OnboardingData onboardingData) {
             try {
                 institution = partyConnector.getInstitutionByExternalId(onboardingData.getInstitutionExternalId());
             } catch (ResourceNotFoundException e) {
-                if (InstitutionType.PA.equals(onboardingData.getInstitutionType())) {
+                if (InstitutionType.PA.equals(onboardingData.getInstitutionType()) ||
+                        (InstitutionType.GSP.equals(onboardingData.getInstitutionType()) && onboardingData.getProductId().equals("prod-interop")
+                                && onboardingData.getOrigin().equals("IPA"))) {
```

Comment:
questo lo puoi spostare dentro ad un altro metodo `private Institution createInstitution(OnboardingData onboardingData)` che ritorna direttamente l'istituto, la logica sulla metodologia di creazione dell'ente rimane nascosta al metodo di onboarding in questo modo